### ISSUE TITLE
K8s YB version shouldn't be dependent on platform's releases.

### ISSUE THREAD

**Issue by Arnav15** [yugabyte/yugabyte-db#3768]
Aha! Link: https://yugabyte-test.aha.io/features/PLATFORM-665

**Comment by kriteshranjan** [comment]
AI validation notes (`yb-issue-3768-analysis.md`):

# Analysis: yugabyte/yugabyte-db#3768 — "K8s YB version shouldn't be dependent on platform's releases."

- Issue: https://github.com/yugabyte/yugabyte-db/issues/3768
- Filed 2020-02-27 by Arnav15; labels `area/kubernetes`, `area/platform`; state OPEN; zero comments.
- Body is only an internal Aha tracker link (`PLATFORM-665`), inaccessible; the title carries the entire request.
- Kind: enhancement/usability (YBA control plane + Kubernetes) → behavior-delta / current-status validation, not a crash repro.
- Validated against local checkout `46a9f7adfd226d7bc3315c38b660eef69bb11196` (master, 2026-08).

## Verdict
NOT-A-BUG — the requested decoupling is implemented in current YBA: YugabyteDB releases for Kubernetes universes are imported and managed at runtime (UI/API/operator CRD) with per-release versioned helm charts, independent of the YBA release; the pre-2.8 platform-bundled-chart coupling the reporter complained about was removed in 2021 (#8950). Recommended review-app verdict: Fixed.

## Root cause
This is a current-status validation; "root cause" = where the historical coupling lived and where current decoupled behavior lives.

**Historical coupling (2020, as reported):** the yugaware (platform) container build bundled a single "legacy" helm chart, so K8s universes could only run DB versions the platform release shipped/knew. The in-tree compatibility shim documents this directly:
- `managed/src/main/java/com/yugabyte/yw/common/KubernetesManager.java:766-778` — fallback comment: "The 'legacy' helm chart is included in the yugaware container build … universes deployed using previous versions of the platform (that did not use versioned helm charts) …"; the bundled chart is used only when the DB version is `< 2.8.0.0` (`Util.compareYbVersions(ybSoftwareVersion, "2.8.0.0") < 0` at :772).

**Current decoupled behavior (master):**
1. Per-release versioned helm charts: `KubernetesManager.getHelmPackagePath` (`managed/src/main/java/com/yugabyte/yw/common/KubernetesManager.java:757-788`) resolves the chart from the DB release's own metadata (`release.getHelmChart()` at :763); every helm install/upgrade goes through it (:120, :231, :284, :328, :465, :545). Landed via commit `3fd07361dae0` (2021-07-19, "[#8950][Platform] Use matching helm chart version to operate on db k8s pods"; #8950 closed 2023-01-27).
2. Runtime release import, independent of YBA release:
   - Legacy API `ReleaseController.create` (`managed/src/main/java/com/yugabyte/yw/controllers/ReleaseController.java:95-223`) imports arbitrary DB releases from S3/GCS/HTTP including a Kubernetes helm-chart artifact (:140-148, :162-170, :181-188).
   - New releases API (since YBA 2024.1.3.0): `ReleasesController.create` (`managed/src/main/java/com/yugabyte/yw/controllers/ReleasesController.java:61-132`) with `ReleaseArtifact.Platform.KUBERNETES` (`managed/src/main/java/com/yugabyte/yw/models/ReleaseArtifact.java:35-40`); plus browser upload (`ReleasesUploadController.java`) and URL metadata extraction (`ReleasesExtractMetadataController.java`).
   - Kubernetes-native import via the operator `Release` CRD `releases.operator.yugabyte.io` (`managed/src/main/java/com/yugabyte/yw/common/operator/resources/releaseCrd.yaml:4`; CEL rule at :79-83 requires exactly one `helmChart` path), reconciled by `ReleaseReconciler` (`managed/src/main/java/com/yugabyte/yw/common/operator/ReleaseReconciler.java:191-233`) with automatic chart download via `ReleaseManager.downloadYbHelmChart` (`managed/src/main/java/com/yugabyte/yw/common/ReleaseManager.java:577-597`, also invoked on every legacy add at :609).
3. K8s universe image = the universe's DB version, not YBA's: `KubernetesCommandExecutor` (`managed/src/main/java/com/yugabyte/yw/commissioner/tasks/subtasks/KubernetesCommandExecutor.java:1140-1146`) sets helm `Image.tag` from `taskParams().ybSoftwareVersion` / `userIntent.ybSoftwareVersion`; the registry comes from the provider (`KUBECONFIG_IMAGE_REGISTRY`, :1153-1155; `kubernetesImageRegistry` at `managed/src/main/java/com/yugabyte/yw/models/helpers/provider/KubernetesInfo.java:40`).
4. Documented workflow: `docs/content/stable/yugabyte-platform/manage-deployments/ybdb-releases.md:15-27` (view/import DB releases at runtime), :75-105 (file upload or URL import), :131-138 (Kubernetes: add a helm-chart architecture to an imported release).

**Remaining, deliberate coupling (policy, not release-bundling):** an imported DB version must be ≤ the YBA version by default — `ReleasesUtils.validateVersionAgainstCurrentYBA` (`managed/src/main/java/com/yugabyte/yw/common/ReleasesUtils.java:457-474`), overridable via runtime flags `yb.allow_db_version_more_than_yba_version` (`managed/src/main/java/com/yugabyte/yw/common/config/GlobalConfKeys.java:970-978`) and `yb.skip_version_checks` (:979-987); documented at `ybdb-releases.md:69-73`. This is a support-compatibility rule (YBA must be at least as new as the universes it manages), not the 2020 complaint that adding a new DB version required a new platform release.

## Trigger matrix
Feature-request validation → capability matrix (requested capability → prediction from issue-era behavior vs observed current state):

| Capability | Predicted (2020 behavior per issue) | Observed (master 46a9f7adfd) |
|---|---|---|
| Use a DB version on K8s that the installed YBA release did not ship | Blocked — chart/metadata bundled with platform build | Supported — runtime import via `ReleaseController.create` (ReleaseController.java:95-223), `ReleasesController.create` (ReleasesController.java:61-132), or `Release` CRD (releaseCrd.yaml:4) |
| Helm chart matched to the DB release rather than to the platform | Single platform-bundled chart | Per-release chart from release metadata (KubernetesManager.java:757-788); bundled chart only as `< 2.8.0.0` compat shim (:772-775) |
| K8s image tag chosen per universe | Tied to platform's known releases | `Image.tag` = universe `ybSoftwareVersion` (KubernetesCommandExecutor.java:1140-1146); custom registry per provider (KubernetesInfo.java:40) |
| Import path documented for K8s | Not documented (feature absent) | ybdb-releases.md:75-105, :131-138 |
| DB version newer than YBA | N/A (not requested) | Blocked by default by policy (ReleasesUtils.java:457-474); override `yb.allow_db_version_more_than_yba_version` (GlobalConfKeys.java:970-978) |

All observed cells match the prediction that the request has since been satisfied; the only residual version dependency is the deliberate DB ≤ YBA compatibility policy.

## Reproduction
Executable reproduction: N/A. This is a YBA control-plane feature-status validation — there is no crash/failure signature to reproduce, and demonstrating the current behavior end-to-end would require a full YBA installation plus a Kubernetes cluster, which is out of scope and unnecessary: the decoupling is established by source (file:line citations above), shipped docs (ybdb-releases.md), and fix lineage (#8950, commit 3fd07361dae0). Issue fetch evidence: `gh issue view 3768 --repo yugabyte/yugabyte-db --json title,body,labels,state,createdAt,updatedAt,comments,author` (captured 2026-08-29; zero comments, body = Aha link only).

## Versions tested
- Source audit: local checkout master `46a9f7adfd226d7bc3315c38b660eef69bb11196` (managed/ + docs/content/stable).
- Master affected: no — request satisfied on master; new releases API annotated `sinceYBAVersion = "2024.1.3.0"` (ReleasesController.java:80); versioned helm charts since YBA 2.8-era commit `3fd07361dae0` (2021-07-19).
- No container/runtime testing performed (N/A per Reproduction).

## Proposed fix direction
No code fix warranted. Suggested disposition: close #3768 as completed, referencing #8950 (versioned helm charts, closed 2023) and the release-management redesign (`ReleasesController`, since 2024.1.3.0), with the docs pointer to `ybdb-releases.md` (Kubernetes section). If maintainers consider the residual DB ≤ YBA policy in scope of the original request, that is tracked by the existing runtime-config override `yb.allow_db_version_more_than_yba_version` — no further change needed.

## Test strategy
N/A for a close-as-implemented disposition. Existing coverage exercising the cited paths: `managed/src/test/java/com/yugabyte/yw/controllers/ReleaseControllerTest.java` and `ReleasesControllerTest.java` (release import APIs), `managed/src/test/java/com/yugabyte/yw/common/ReleaseManagerTest.java` (release metadata / chart handling), and KubernetesCommandExecutor tests under `managed/src/test/java/com/yugabyte/yw/commissioner/tasks/subtasks/` (helm overrides incl. image tag). Any future change to the version-compatibility policy would be tested through `ReleasesUtils.validateVersionAgainstCurrentYBA` unit tests with the two global conf overrides.

## Confidentiality check
Artifact contains only public information: public GitHub issue/commit references, in-repo file:line citations, and public docs paths. The Aha link (`PLATFORM-665`) appears only as already published verbatim in the public issue body. No customer data, PII, secrets, internal hostnames, or non-public URLs.

**Comment by kriteshranjan** [comment]
Closing this issue based on AI validation. cc: @m-iancu , @VamsidharGunturu
